jax_galsim/gaussian.py

```diff
@@ -0,0 +1,133 @@
+        elif half_light_radius is not None:
+            if sigma is not None:
+                raise _galsim.GalSimIncompatibleValuesError(
+                    "Only one of sigma, fwhm, and half_light_radius may be specified",
+                    fwhm=fwhm,
+                    sigma=sigma,
+                    half_light_radius=half_light_radius,
+                )
+            else:
+                super().__init__(
+                    half_light_radius=half_light_radius, flux=flux, gsparams=gsparams
+                )
+        elif sigma is None:
+            raise _galsim.GalSimIncompatibleValuesError(
+                "One of sigma, fwhm, and half_light_radius must be specified",
+                fwhm=fwhm,
+                sigma=sigma,
+                half_light_radius=half_light_radius,
+            )
+        else:
+            super().__init__(sigma=sigma, flux=flux, gsparams=gsparams)
+
+        self._sigsq = self.sigma**2
+        self._inv_sigsq = 1.0 / self._sigsq
+        self._norm = self.flux * self._inv_sigsq * Gaussian._inv_twopi
+
+    @property
+    def sigma(self):
+        """The sigma of this Gaussian profile"""
+        if "fwhm" in self.params:
+            return self.params["fwhm"] / Gaussian._fwhm_factor
+        elif "half_light_radius" in self.params:
+            return self.params["half_light_radius"] / Gaussian._hlr_factor
+        else:
+            return self.params["sigma"]
+
+    @property
+    def half_light_radius(self):
+        """The half-light radius of this Gaussian profile"""
+        return self.sigma * Gaussian._hlr_factor
+
+    @property
+    def fwhm(self):
+        """The FWHM of this Gaussian profile"""
+        return self.sigma * Gaussian._fwhm_factor
+
+    def __hash__(self):
+        return hash(("galsim.Gaussian", self.sigma, self.flux, self.gsparams))
+
+    def __repr__(self):
+        return "galsim.Gaussian(sigma=%r, flux=%r, gsparams=%r)" % (
+            self.sigma,
+            self.flux,
+            self.gsparams,
+        )
+
+    def __str__(self):
+        s = "galsim.Gaussian(sigma=%s" % self.sigma
+        if self.flux != 1.0:
+            s += ", flux=%s" % self.flux
+        s += ")"
+        return s
+
+    @property
+    def _maxk(self):
+        return jnp.sqrt(-2.0 * jnp.log(self.gsparams.maxk_threshold)) / self.sigma
+
+    @property
+    def _stepk(self):
+        R = jnp.array(
+            [
+                jnp.sqrt(-2.0 * jnp.log(self.gsparams.folding_threshold)),
+                self.gsparams.stepk_minimum_hlr * Gaussian._hlr_factor,
+            ]
+        ).max()
+        return jnp.pi / (R * self.sigma)
+
+    @property
+    def _max_sb(self):
+        return self._norm
+
+    def _xValue(self, pos):
+        rsq = pos.x**2 + pos.y**2
+        return self._norm * jnp.exp(-0.5 * rsq * self._inv_sigsq)
+
+    def _kValue(self, kpos):
+        ksq = (kpos.x**2 + kpos.y**2) * self._sigsq
+        return self.flux * jnp.exp(-0.5 * ksq)
```

Comment:
This is quite minor: I noticed for this function you use `self.flux` instead of `self._flux` like in the galsim implementation of Gaussian. In the Exponential class you use `self._flux` though, is there a reason to deviate from the original galsim implementation?

Reply:
yeah, the reasonning there is to make sure that any objet parameters we use in the class come directly from the self.params dictionary, to avoid potential confusions and storing the information in multiple places. self.flux is defined in GSObject and is equivalent to self.params['flux'] there is no self._flux

Reply:
thanks, that makes sense now